packages/eslint-plugin/tests/rules/thenable-in-promise-aggregators.test.ts

```diff
@@ -0,0 +1,711 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/thenable-in-promise-aggregators';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('thenable-in-promise-aggregators', rule, {
+  valid: [
```

Comment:
[Testing] Some additional valid cases to put in...

* <code>Promise[\`all\`]</code>
* `Promise["all"]`
* `const key = "all"; Promise[key]`

Reply:
I don't know how to test `Promise["all"]` without `yarn lint --fix` just swapping it to single quotes, assuming you mean for double quotes. I do have the single quotes case in there near the top of the valid list.

I added the template literal case for both valid/invalid. For invalid it actually failed that test case, I added the necessary code to handle it in the refactored `getMemberName` but I do wonder how bad the performance will be on that since I imagine member expression calls are quite common, though I don't know about indexed ones with non-literals. The only case where this type of thing might be convenient is if there were a parameter to switch between `any` vs `race` and they chose to do it like this:

```ts
async function waitWithTimeout(a: Promise<T>, aggregator: 'race' | 'any') {
  const timeout = new Promise((resolve) => setTimeout(resolve, 100));
  await Promise[aggregator]([a, timeout])
}
```

I added tests for this type of thing, but it meant that `getMemberName` had to become `isMemberName` instead to handle unions

Reply:
> I don't know how to test `Promise["all"]` without `yarn lint --fix` just swapping it to single quotes, assuming you mean for double quotes

Oh yeah search `noFormat`, we've hit that annoyance before.